src/coreComponents/denseLinearAlgebra/unitTests/testDenseLASolvers.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2016-2024 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2024 Total, S.A
+ * Copyright (c) 2018-2024 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2024 Chevron
+ * Copyright (c) 2019-     GEOS/GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#include "denseLinearAlgebra/denseLASolvers.hpp"
+
+// TPL includes
+#include <gtest/gtest.h>
+
+#include <random>
+
+
+using namespace geos;
+
+
+constexpr real64 machinePrecision = 1.0e2 * LvArray::NumericLimits< real64 >::epsilon;
+
+template< std::ptrdiff_t N >
+struct LinearSystem
+{
+  static constexpr std::ptrdiff_t size() { return N; }
+
+  LinearSystem( short int seed )
+  {
+    std::mt19937 generator( seed ); 
+    std::uniform_real_distribution< real64 > distribution( -10.0, 10.0 );
+    for( ptrdiff_t i=0; i<N; ++i )
+    {
+      solution[i] = distribution( generator );
+      for( ptrdiff_t j=0; j<N; ++j )
+      {
+        matrix[i][j] =  distribution( generator );
+      }
```

Comment:
```suggestion
      for( ptrdiff_t j=0; j<N; ++j )
      {
        matrix[i][j] =  distribution( generator );
      }
      matrix[i][i] += 1.0e2 * machinePrecision;
```

Extremely unlikely, but in principle a singular matrix could be generated. I would add a perturbation to the diagonal to ensure it is nonsingular.